t/plans.t

```diff
@@ -0,0 +1,106 @@
+=pod
+
+=encoding utf-8
+
+=head1 PURPOSE
+
+Tests for various plans
+
+=cut
+
+use v5.14;
+use autodie;
+use utf8;
+use feature "state";
+use Test::Modern;
+use Test::Exception;
+use Digest::SHA qw(sha1_hex);
+
+use Attean;
+use Attean::RDF;
+
+use Attean::Plan::Iterator;
+
+my $ci = Attean::CodeIterator->new(
+											  generator => sub {
+												  state $i = 0;
+												  return undef if ($i > 2);
+												  return Attean::Result->new(bindings => { 'o' => literal($i) });
+											  },
+											  item_type => 'Attean::API::Result',
+											  variables => ['o']
+											 );
+
+isa_ok($ci, 'Attean::CodeIterator');
+
+my @values = map { Attean::Result->new(bindings => { 'o' => literal($_) }) } (1,2,3);
+my $li = Attean::ListIterator->new(
+  values => \@values,
+  item_type => 'Attean::API::Result',
+  variables => ['o']
+);
+
+isa_ok($li, 'Attean::ListIterator');
+
+#subtest 'CodeIterator without size' => sub
+{
+	my $plan = Attean::Plan::Iterator->new( variables => [variable('o')],
+														 iterator => $ci,
+														 distinct => 0,
+														 ordered => [] );
+	isa_ok($plan, 'Attean::Plan::Iterator');
+	does_ok($plan, 'Attean::API::Plan');
+	can_ok($plan, 'iterator');
+	ok(! $plan->has_size_estimate, 'Has no size estimate');
+	is($plan->as_string, "- Iterator (?o)\n", 'Correct serialization');
+};
+
+#subtest 'CodeIterator with size' => sub
+{
+	my $plan = Attean::Plan::Iterator->new( variables => [variable('o')],
+														 iterator => $ci,
+														 distinct => 0,
+														 size_estimate => 2,
+														 ordered => [] );
+	isa_ok($plan, 'Attean::Plan::Iterator');
+	does_ok($plan, 'Attean::API::Plan');
+	can_ok($plan, 'iterator');
+	ok($plan->has_size_estimate, 'Has size estimate');
+	is($plan->size_estimate, 2, 'Correct returned estimate');
+	is($plan->as_string, "- Iterator (?o with 2 elements)\n", 'Correct serialization');
+};
+
+{
+	my $plan = Attean::Plan::Iterator->new( variables => [variable('o')],
+														 iterator => $li,
+														 distinct => 0,
+														 ordered => [] );
+	isa_ok($plan, 'Attean::Plan::Iterator');
+	does_ok($plan, 'Attean::API::Plan');
+	can_ok($plan, 'iterator');
+ TODO: {
```

Comment:
If we're OK with the predicate only being true after the attribute is accessed, can we get rid of the TODO here? I don't have strong feelings about how this is settled, but I'm wary of merging TODO tests without a compelling reason.